cli/src/services/api.service.ts

```diff
@@ -1,56 +1,106 @@
 import {
-  AlbumApi,
-  APIKeyApi,
-  AssetApi,
-  AuthenticationApi,
-  Configuration,
-  JobApi,
-  OAuthApi,
-  ServerInfoApi,
-  SystemConfigApi,
-  UserApi,
+  addAssetsToAlbum,
+  checkBulkUpload,
+  createAlbum,
+  createApiKey,
+  getAllAlbums,
+  getAllAssets,
+  getAssetStatistics,
+  getMyUserInfo,
+  getServerVersion,
+  getSupportedMediaTypes,
+  login,
+  pingServer,
+  signUpAdmin,
+  uploadFile,
+  ApiKeyCreateDto,
+  AssetBulkUploadCheckDto,
+  BulkIdsDto,
+  CreateAlbumDto,
+  CreateAssetDto,
+  LoginCredentialDto,
+  SignUpDto,
 } from '@immich/sdk';
 
+/**
+ * Wraps the underlying API to abstract away the options and make API calls mockable for testing.
+ */
 export class ImmichApi {
-  public userApi: UserApi;
-  public albumApi: AlbumApi;
-  public assetApi: AssetApi;
-  public authenticationApi: AuthenticationApi;
-  public oauthApi: OAuthApi;
-  public serverInfoApi: ServerInfoApi;
-  public jobApi: JobApi;
-  public keyApi: APIKeyApi;
-  public systemConfigApi: SystemConfigApi;
-
-  private readonly config;
+  private readonly options;
 
   constructor(
     public instanceUrl: string,
     public apiKey: string,
   ) {
-    this.config = new Configuration({
-      basePath: instanceUrl,
+    this.options = {
+      baseUrl: instanceUrl,
       headers: {
         'x-api-key': apiKey,
       },
-    });
-
-    this.userApi = new UserApi(this.config);
-    this.albumApi = new AlbumApi(this.config);
-    this.assetApi = new AssetApi(this.config);
-    this.authenticationApi = new AuthenticationApi(this.config);
-    this.oauthApi = new OAuthApi(this.config);
-    this.serverInfoApi = new ServerInfoApi(this.config);
-    this.jobApi = new JobApi(this.config);
-    this.keyApi = new APIKeyApi(this.config);
-    this.systemConfigApi = new SystemConfigApi(this.config);
+    };
   }
 
   setApiKey(apiKey: string) {
     this.apiKey = apiKey;
-    if (!this.config.headers) {
+    if (!this.options.headers) {
       throw new Error('missing headers');
     }
-    this.config.headers['x-api-key'] = apiKey;
+    this.options.headers['x-api-key'] = apiKey;
+  }
+
+  async addAssetsToAlbum(id: string, bulkIdsDto: BulkIdsDto) {
+    return await addAssetsToAlbum({ id, bulkIdsDto }, this.options);
+  }
```

Comment:
you can remove `async` and `await` from all of these methods. I prefer not having any of those unless they're actually needed.